protocol/messenger_delete_message_test.go

```diff
@@ -23,6 +23,7 @@ func (s *MessengerDeleteMessageSuite) TestDeleteMessage() {
 	theirMessenger := s.newMessenger()
 	_, err := theirMessenger.Start()
 	s.Require().NoError(err)
+	defer theirMessenger.Shutdown() // nolint: errcheck
```

Comment:
I wonder if it is possible to customise the linter to ignore errors from a deferred function

Reply:
I think that might not be what we want, in tests or like in this case, it makes sense, but people might do it in places where they shouldn't I guess

Reply:
Having a quick look through this commit https://github.com/amnonbc/errcheck/commit/3df4ffcb5f4958171ceb8eb833574522560ed9ae#diff-4adfb6e9793bcec46facbdb73ff436ac5f2b9a060577c73167801a8812a8fb69 what I'm suggesting appears to be possible but would require changes to the `errcheck` source code.

`errcheck` can be configured to ignore all test files entirely but doesn't have the logic to only conditionally ignore defer functions that are in test files. Seems possible by updating the `visitor` state and checking if the `visitor` is in a test file and is inspecting a defer statement. But this functionality is not configurable from the config settings alone.